x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawl_detail_logic.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { CrawlRequest, CrawlRequestFromServer } from './types';
+import { crawlRequestServerToClient } from './utils';
+
+export interface CrawlDetailValues {
+  dataLoading: boolean;
+  isFlyoutOpen: boolean;
+  request: CrawlRequest | null;
+}
+
+interface CrawlDetailActions {
+  closeFlyout(): void;
+  fetchCrawlRequest(requestId: string): { requestId: string };
+  onRecieveCrawlRequest(request: CrawlRequest): { request: CrawlRequest };
+}
+
+export const CrawlDetailLogic = kea<MakeLogicType<CrawlDetailValues, CrawlDetailActions>>({
+  path: ['enterprise_search', 'app_search', 'crawl_detail_logic'],
+  actions: {
+    closeFlyout: true,
+    fetchCrawlRequest: (requestId) => ({ requestId }),
+    onRecieveCrawlRequest: (request) => ({ request }),
+  },
+  reducers: {
+    dataLoading: [
+      true,
+      {
+        fetchCrawlRequest: () => true,
+        onRecieveCrawlRequest: () => false,
+      },
+    ],
+    request: [
+      null,
+      {
+        onRecieveCrawlRequest: (_, { request }) => request,
+      },
+    ],
+    isFlyoutOpen: [
+      false,
+      {
+        fetchCrawlRequest: () => true,
+        closeFlyout: () => false,
+      },
+    ],
+  },
+  listeners: ({ actions }) => ({
+    fetchCrawlRequest: async ({ requestId }) => {
```

Comment:
I think it would have been cleaner to decouple opening the modal and fetching data.

So we'd have a discrete action that corresponds to opening the modal, like `openModal`, and then trigger `fetchCrawlRequest` in a `useEffect` inside of the modal, for instance.

Not a blocker for this PR, just thought I'd mention it.

Reply:
I could see a use case for creating a seperate `openModal` action, seperate from a `fetchCrawlRequest` action, but I'd rather trigger those two in sequence through use of a listener attached to the `openModal` action that would call `actions.fetchCrawlRequest()` instead of going through a `useEffect`.

Reply:
Yep, that would totally work as well. This is a pretty self contained logic so I think it's fine as is.

Reply:
Decoupling is useful for situations where we're like, I really want to fetch details for this *other* screen that doesn't have a modal and I don't really want to start from scratch, but I can't use this logic because it's tightly coupled with this modal property.